### PR TITLE
refactor(sbb-card): lit migration

### PR DESCRIPTION
**Pull request by dauriamarco**
Migrate `sbb-card`, `sbb-card-badge` and `sbb-card-action` to Lit.

### CODE REVIEW

File: src/components/sbb-card-action/sbb-card-action.e2e.ts [hunk 1]
```diff
@@ -1,222 +1,223 @@
...
+    const buttons = document.querySelectorAll('button');
+    expect(buttons.length).to.be.equal(2);
+    expect(
+      Array.from(buttons).every((el) => el.getAttribute('data-card-focusable') !== null),
+    ).to.be.equal(true);
 
     // Remove all buttons
-    await page.evaluate(() => document.querySelectorAll('button').forEach((el) => el.remove()));
-    await page.waitForChanges();
+    document.querySelectorAll('button').forEach((el) => el.remove());
```
Comment (jeripeierSBB) [inline src/components/sbb-card-action/sbb-card-action.e2e.ts:123]: can we not reuse "buttons" const ?

File: src/components/sbb-card-action/sbb-card-action.e2e.ts [hunk 2]
```diff
@@ -1,222 +1,223 @@
...
       await waitForCondition(() => changeSpy.events.length === 1);
-      expect(changeSpy).toHaveReceivedEventTimes(1);
+      expect(changeSpy.count).to.be.equal(1);
     });
 
     it('should dispatch click event on pressing Enter', async () => {
-      const changeSpy = await page.spyOnEvent('click');
-      await element.press('Enter');
-      expect(changeSpy).toHaveReceivedEvent();
+      const changeSpy = new EventSpy('click');
```
Comment (jeripeierSBB) [inline src/components/sbb-card-action/sbb-card-action.e2e.ts:183]:
```suggestion
      const clickSpy = new EventSpy('click');
```

File: src/components/sbb-card-action/sbb-card-action.e2e.ts [hunk 3]
```diff
@@ -1,222 +1,223 @@
...
+      action.focus();
+      await sendKeys({ press: 'Enter' });
+      expect(changeSpy.count).to.be.greaterThan(0);
     });
 
     it('should dispatch click event on pressing Space', async () => {
-      const changeSpy = await page.spyOnEvent('click');
-      await element.press(' ');
-      expect(changeSpy).toHaveReceivedEvent();
+      const changeSpy = new EventSpy('click');
```
Comment (jeripeierSBB) [inline src/components/sbb-card-action/sbb-card-action.e2e.ts:190]:
```suggestion
      const clickSpy = new EventSpy('click');
```

File: src/components/sbb-card-action/sbb-card-action.e2e.ts [hunk 4]
```diff
@@ -1,222 +1,223 @@
...
 
     it('should dispatch click event on pressing Enter with href', async () => {
       element.setAttribute('href', 'test');
-      await page.waitForChanges();
+      await element.updateComplete;
 
-      const changeSpy = await page.spyOnEvent('click');
-      await element.press('Enter');
-      expect(changeSpy).toHaveReceivedEvent();
+      const changeSpy = new EventSpy('click');
```
Comment (jeripeierSBB) [inline src/components/sbb-card-action/sbb-card-action.e2e.ts:200]:
```suggestion
      const clickSpy = new EventSpy('click');
```

File: src/components/sbb-card-action/sbb-card-action.e2e.ts [hunk 5]
```diff
@@ -1,222 +1,223 @@
...
     it('should not dispatch click event on pressing Space with href', async () => {
-      element.setAttribute('href', 'test');
-      await page.waitForChanges();
+      action.setAttribute('href', 'test');
+      await element.updateComplete;
 
-      const changeSpy = await page.spyOnEvent('click');
-      await element.press(' ');
-      expect(changeSpy).not.toHaveReceivedEvent();
+      const changeSpy = new EventSpy('click');
```
Comment (jeripeierSBB) [inline src/components/sbb-card-action/sbb-card-action.e2e.ts:210]:
```suggestion
      const clickSpy = new EventSpy('click');
```

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

File: src/components/sbb-card-action/sbb-card-action.tsx [hunk 6]
```diff
@@ -28,75 +14,84 @@ import {
...
   private _handlerRepository = new HandlerRepository(
-    this._element,
+    this,
     actionElementHandlerAspect,
     languageChangeHandlerAspect((l) => (this._currentLanguage = l)),
   );
 
-  public constructor() {
-    // Set slot name as early as possible
-    this._element.setAttribute('slot', 'action');
```
[the remark this answers is not in the document]
Comment (dauriamarco) [reply src/components/sbb-card-action/sbb-card-action.tsx:93]: Unfortunately it was simply not working, so I had to move it

File: src/components/sbb-card-badge/sbb-card-badge.tsx [hunk 7]
```diff
@@ -1,52 +1,57 @@
...
-  /** Color of the card badge. */
-  @Prop({ reflect: true }) public color: InterfaceSbbCardBadgeAttributes['color'] = 'charcoal';
-
-  @Element() private _element!: HTMLElement;
+@customElement('sbb-card-badge')
+export class SbbCardBadge extends LitElement {
+  public static override styles: CSSResult = Style;
 
-  public constructor() {
-    // Set slot name as early as possible
```
[the remark this answers is not in the document]
Comment (dauriamarco) [reply src/components/sbb-card-badge/sbb-card-badge.tsx:22]: Same here, it wasn't working and I had to move it

File: src/components/sbb-card/sbb-card.spec.ts [hunk 8]
```diff
@@ -20,9 +19,8 @@ describe('sbb-card', () => {
           <span>19.99</span>
         </sbb-card-badge>
       </sbb-card>`,
-    });
+    );
 
-    expect(root.shadowRoot.querySelector(cardBadgeWrapperSelector)).toBeFalsy();
-    expect(root).not.toHaveAttribute(hasBadgeDataAttributeName);
```
[the remark this answers is not in the document]
Comment (dauriamarco) [reply src/components/sbb-card/sbb-card.spec.ts:26]: I thought this was an error, as the attribute remains on the host even for smaller sizes. I checked the behaviour of the current implementation on master, so I wonder why it now passes on master?

File: src/components/sbb-card-action/sbb-card-action.e2e.ts
Comment (TomMenga) [inline src/components/sbb-card-action/sbb-card-action.e2e.ts:1]: Should we try to separate spec and e2e tests?
The effort and impact should be minimal.
@jeripeierSBB @dauriamarco what do you think?

Comment (jeripeierSBB) [reply src/components/sbb-card-action/sbb-card-action.e2e.ts:1]: For me it doesn't matter. maybe we will once anyways merge the two files for every component :)

**Review by jeripeierSBB — APPROVED** [review]
LGTM, yay!

**Pull request merged**